core/src/main/java/com/linecorp/armeria/server/ServiceConfigBuilder.java

```diff
@@ -302,18 +305,46 @@ ServiceConfig build(ServiceNaming defaultServiceNaming,
                                                                      unhandledExceptionsReporter);
         }
 
+        final boolean webSocket = service.as(WebSocketService.class) != null;
```

Comment:
These changes look as is. 
Optionally, if there is a desire to provide default settings based on the service type,
it would be better to add `ServiceOptions` to `HttpService` like `RequestOptions`.

```
interface HttpService {

  default ServiceOptions defaultOptions() {
    return ServiceOptions.of();
  }
}

class WebSocketService {
    @Override
    default ServiceOptions defaultOptions() {
       return WEB_SOCKET_DEFAULT_OPTIONS;
    }
}

// ServiceConfigBuilder

if (service.option() != ServiceOptions.of()) {
   // override the default values with service.options().
}
```

Reply:
That's a good suggestion. 👍

Reply:
Sure, I will.